src/array.c

```diff
@@ -150,7 +152,18 @@ SDB_API int sdb_array_insert(Sdb *s, const char *key, int idx, const char *val,
 	// we can optimize this by caching value len in memory . add
 	// sdb_const_get_size()
 	lstr = strlen (str); 
-	x = malloc (lval + lstr + 2);
+
+#define SIZE_ADD_OVFCHK(x, y) ((SIZE_MAX - (x)) <= y)
```

Comment:
this should be defined in .h and checked by a guard to avoid double definition in r2land